src/lightning/pytorch/trainer/connectors/accelerator_connector.py

```diff
@@ -159,6 +160,11 @@ def __init__(
         # 4. Instantiate Strategy - Part 1
         if self._strategy_flag == "auto":
             self._strategy_flag = self._choose_strategy()
+
+        # Change fsdp to xla_fsdp if using TPU
+        if self._strategy_flag == "fsdp" and self._accelerator_flag == "tpu":
+            self._strategy_flag = "xla_fsdp"
```

Comment:
This logic would fit well inside the `_check_strategy_and_fallback` method

Reply:
Good idea, I will make the change